sphinx/ext/autodoc/__init__.py

```diff
@@ -418,9 +419,12 @@ def format_signature(self):
         else:
             return ''
 
-    def add_directive_header(self, sig):
-        # type: (str) -> None
+    def add_directive_header(self, sig, additional_options=None):
```

Comment:
How about renaming `additional_options` to `options` simply?

Reply:
In addition, how about giving options via keyword argument?
I think `add_directive_header(sig, abstract=True)` is much better. Passing options via both argument and `self.option` is a bit hard to understand to me.

Reply:
As mentioned below, that argument should have better been named `revealed_options`.
I think I've come up with a way better implementation of what I've wanted to achieve. Could you check the new diff and see if it is understandable to you?

The main advantage of only telling `Documenter.add_directive_header` which parts of the `options` dict to be included in the ReST ouput is that extending becomes easier. For instance, to add an `:async:` flag, you don't need to introduce a new argument to `add_directive_header`, you can instead just set the `async` key in the options dict and add it to the `_revealed_header_options` whitelist. Changing a signature for that would be more effort for something that doesn't really change functionality.

Reply:
The relevant change is in https://github.com/sphinx-doc/sphinx/pull/6138/commits/9c58354de8c76ef1b3587a6caaf1b7e248e87354.